### PR TITLE
CLOUDP-271316: Rename import to standard apierrors

### PR DESCRIPTION
**Pull request by josvazg**
### All Submissions:

* [X] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?

### CODE REVIEW

File: pkg/controller/connectionsecret/ensuresecret.go [hunk 1]
```diff
@@ -6,7 +6,7 @@ import (
 	"net/url"
 
 	corev1 "k8s.io/api/core/v1"
-	apiErrors "k8s.io/apimachinery/pkg/api/errors"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
```
Comment (cveticm) [inline pkg/controller/connectionsecret/ensuresecret.go:9]: Question: What is the rational for changing this?

**Review by cveticm — CHANGES_REQUESTED** [review]
Please associate a ticket to this PR

File: pkg/controller/connectionsecret/ensuresecret.go (hunk 1 above)
Comment (josvazg) [reply pkg/controller/connectionsecret/ensuresecret.go:9]: It comes from a comment by @s-urbaniak  here:
https://github.com/mongodb/mongodb-atlas-kubernetes/pull/1798#discussion_r1741512235

I decided to make it a separate PR

Comment (cveticm) [reply pkg/controller/connectionsecret/ensuresecret.go:9]: Ok, so we want to keep with the naming convention set by k8s. This new style is less readable but I agree consistency is better.

Can you create a ticket to associate this with?

**Comment by josvazg** [comment]
CLOUDP-271316

File: pkg/controller/connectionsecret/ensuresecret.go (hunk 1 above)
Comment (josvazg) [reply pkg/controller/connectionsecret/ensuresecret.go:9]: Done CLOUDP-271316

Comment (josvazg) [reply pkg/controller/connectionsecret/ensuresecret.go:9]: BTW `apiErrors` is camel case and not considered idiomatic as a package alias name:
https://go.dev/blog/package-names

Comment (cveticm) [reply pkg/controller/connectionsecret/ensuresecret.go:9]: Aha, noted, thanks for pointing me to that!

LGTM from me then

**Review by cveticm — APPROVED** [review]

**Review by s-urbaniak — APPROVED** [review]

**Pull request merged**